packages/components/src/internal/url/URLResolver.ts

```diff
@@ -513,6 +513,14 @@ export const URL_MAPPERS = {
 
 export class URLResolver {
     private mapURL = (mapper: MapURLOptions): string => {
+
+        // Don't override URLs if the URL has a different container than the current container and is not in the subfolder
+        // tree of the current container. This scopes the apps to their current container and sub-containers, and supports
+        // adding FKs from other containers and preserving the URL from the server.
+        let urlContainerPath = parsePathName(mapper.url).containerPath;
```

Comment:
Should probably be defensive against an empty URL on the `mapper.url` here because `parsePathName` is not defensive. Something like:

```ts
const urlContainerPath = parsePathName(mapper.url ?? '').containerPath;
```

nit: Should be declared as a `const`.

Reply:
I just put the whole block in if statement for empty url so it can go on to URLResolver if it's blank.